app/models/exchange_rate.rb

```diff
@@ -12,11 +12,11 @@ def find_rate(from:, to:, date:)
     end
 
     def find_rate_or_fetch(from:, to:, date:)
-      find_rate(from:, to:, date:) || fetch_rate_from_provider(from:, to:, date:).tap(&:save!)
+      find_rate(from:, to:, date:) || fetch_rate_from_provider(from:, to:, date:)&.tap(&:save!)
     end
 
-    def get_rate_series(from, to, date_range)
-      where(base_currency: from, converted_currency: to, date: date_range).order(:date)
+    def get_rates(from, to, dates)
```

Comment:
Renamed the method as it can be used not just for a continuous series of dates, but also for an arbitrary array of dates

Reply:
Great point, I always forget about this 🙇